### ISSUE TITLE
There might be a problem with the project dependency tree. It is likely not a bug in Create React App, but something you need to fix locally.

### ISSUE THREAD

**Issue by ravi19999** [trufflesuite/truffle#2760]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.
Everytime I try to unbox react through truffle, the following message appears, I've tried everything it says but the problem still persists.

```
> client@0.1.0 start C:\users\uddhav\desktop\w\client
> react-scripts start


There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "eslint": "^6.1.0"

Don't try to install it manually: your package manager does it automatically.
However, a different version of eslint was detected higher up in the tree:

  C:\Users\Uddhav\node_modules\eslint (version: 5.16.0)

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "eslint" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if C:\Users\Uddhav\node_modules\eslint is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls eslint in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed eslint.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.

P.S. We know this message is long but please read the steps above :-) We hope you find them helpful!

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! client@0.1.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the client@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Uddhav\AppData\Roaming\npm-cache\_logs\2020-01-15T03_47_18_452Z-debug.log
```

## Steps to Reproduce
truffle unbox react
Please provide the shortest amount of steps to reproduce your issue.

## Expected Behavior
unbox successful, sweet

What you expected to happen.

## Actual Results
There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.
What actually happened. Please give examples and support it with screenshots, copied output or error messages.

## Environment

* Operating System: Windows
* Ethereum client:
* Truffle version (`truffle version`): 5.1.8
* node version (`node --version`): 12.14.1
* npm version (`npm --version`): 6.13.8

**Comment by cds-amal** [comment]
Hi @ravi19999, what version of windows are you using? If you're on windows 10, do you have [developer mode enabled](https://docs.microsoft.com/en-us/windows/uwp/get-started/enable-your-device-for-development)?

**Comment by gnidan** [comment]
Closing this for issue maintenance. Let us know if you want us to get :eyes: on this again and we'll be happy to re-open. Cheers!
